src/app/content/utils/archiveTreeUtils.ts

```diff
@@ -137,10 +137,11 @@ export const archiveTreeSectionIsUnit = (section: LinkedArchiveTreeNode) =>
   isArchiveTree(section)
   && !!section.parent
   && archiveTreeSectionIsBook(section.parent)
-  && getArchiveTreeSectionNumber(section) === null
+  && section.contents.every((node) => (node as LinkedArchiveTree).contents !== undefined)
```

Comment:
Do the other types need changes? It feels like if they were correct you would need an another guard here instead of casting the type.

Reply:
`section.contents` may have items of type `ArchiveTree` or `ArchiveTreePage` and since `ArchiveTreeNode` does not have `contentes` then typescript is not allowing us to check `node.contents` even if `node` is an `object`
I was thinking about doing something like `section.contents.every(archiveTreeSectionIsChapter)` but this will check for a lot more of things than simply having `contents` and I don't know if this is worth it

From the other side, I think that maybe it's worth it to reconsider naming of types and interfaces that we use and simplify them somehow to use `book`, `page`, `unit` and `chapter` instead of `LinkedArchiveTree/Section/Node`
Maybe:
`LinkedArchiveBook` for `book`
`LinkedArchivePage` for `page`
`LinkedArchiveUnit` for `unit`
`LinkedArchiveChapter` for `chapter`

However, I didn't want to do that in this pr without a discussion with Tom, so I've decided to go with a cast here

Reply:
But the type `LinkedArchiveTree` has contents typed as always being there. I think combining guards is a safe way of doing it. 
> if this is worth it

i feel like if the checking for what is desired, then guards are the right place where we should trade slightly more things to check for better type safety.

Reply:
After some investigation:
`node` from `section.contents.every(node => ...)` in `archiveTreeSectionIsUnit` is of type `ArchiveTree` | `ArchiveTreeNode` so if I would like to use it as `section.contents.every(archiveTreeSectionIsChapter)` I would have to type `archiveTreeSectionIsChapter` to allow `ArchiveTree` and `ArchiveTreeNode` but then `!archiveTreeSectionIsBook(section)` will always return `true` because these types does not have `parent` - they are not `linked`
In other terms - If we would like to introduce any changes to this guards I would suggest that we talk about it, prepare a bigger solution and then talk with @TomWoodward about implementation. I'm not sure if this should be done in this pr

Reply:
I've replaced casts with `isArchiveTree` guard that also has cast but at least this is now in one place